overwatch-derive/src/lib.rs

```diff
@@ -152,7 +152,7 @@ fn generate_new_impl(fields: &Punctuated<Field, Comma>) -> proc_macro2::TokenStr
     });
 
     quote! {
-        fn new(settings: Self::Settings, overwatch_handle: ::overwatch_rs::overwatch::handle::OverwatchHandle) -> Self {
+        fn new(settings: Self::Settings, overwatch_handle: ::overwatch_rs::overwatch::handle::OverwatchHandle) -> ::std::result::Result<Self, ::std::boxed::Box<dyn ::std::error::Error + ::std::marker::Send + ::std::marker::Sync + 'static>> {
```

Comment:
This will be simpler with the alias I think :)

Btw, do you think we could make it so we can aggregate the errors instead of returning the first one encountered?

Reply:
@Zeegomo what do you think about that ^^^?

Reply:
I think it could be helpful for sure, mostly depends on how much of an effort it is to do so